src/pages/settings/Wallet/ReportVirtualCardFraudPage.tsx

```diff
@@ -13,61 +13,54 @@ import useThemeStyles from '@hooks/useThemeStyles';
 import * as CardUtils from '@libs/CardUtils';
 import * as ErrorUtils from '@libs/ErrorUtils';
 import Navigation from '@libs/Navigation/Navigation';
+import type {PublicScreensParamList} from '@libs/Navigation/types';
 import NotFoundPage from '@pages/ErrorPage/NotFoundPage';
 import * as Card from '@userActions/Card';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
-import assignedCardPropTypes from './assignedCardPropTypes';
+import type SCREENS from '@src/SCREENS';
+import type {ReportVirtualCardFraudForm} from '@src/types/form';
+import type {Card as OnyxCard} from '@src/types/onyx';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
 
-const propTypes = {
-    /* Onyx Props */
-    formData: PropTypes.shape({
-        isLoading: PropTypes.bool,
-    }),
-    cardList: PropTypes.objectOf(assignedCardPropTypes),
-    /** The parameters needed to authenticate with a short-lived token are in the URL */
-    route: PropTypes.shape({
-        /** Each parameter passed via the URL */
-        params: PropTypes.shape({
-            /** Domain string */
-            domain: PropTypes.string,
-        }),
-    }).isRequired,
-};
+type ReportVirtualCardFraudPageOnyxProps = {
+    /** Form data propTypes */
+    formData: OnyxEntry<ReportVirtualCardFraudForm>;
 
-const defaultProps = {
-    cardList: {},
-    formData: {},
+    /** Card list propTypes */
+    cardList: OnyxEntry<Record<string, OnyxCard>>;
 };
 
+type ReportVirtualCardFraudPageProps = ReportVirtualCardFraudPageOnyxProps & StackScreenProps<PublicScreensParamList, typeof SCREENS.TRANSITION_BETWEEN_APPS>;
+
 function ReportVirtualCardFraudPage({
     route: {
-        params: {domain},
+        params: {domain = ''},
     },
     cardList,
     formData,
-}) {
+}: ReportVirtualCardFraudPageProps) {
     const styles = useThemeStyles();
     const {translate} = useLocalize();
 
     const domainCards = CardUtils.getDomainCards(cardList)[domain];
-    const virtualCard = _.find(domainCards, (card) => card.isVirtual) || {};
-    const virtualCardError = ErrorUtils.getLatestErrorMessage(virtualCard) || '';
+    const virtualCard = domainCards?.find((card) => card.isVirtual);
+    const virtualCardError = ErrorUtils.getLatestErrorMessage(virtualCard?.errors ?? {});
```

Comment:
Changing the parameter of `getLatestErrorMessage` from `virtualCard` to `virtualCard?.errors ?? {}` causes an issue where the server error does not display. This happens because the updated parameter always returns an empty string more context on this Issue https://github.com/Expensify/App/issues/49010#issuecomment-2345118755.